source/slang/slang-emit-c-like.cpp

```diff
@@ -2592,18 +2592,35 @@ void CLikeSourceEmitter::emitRegion(Region* inRegion)
                 //
                 if (auto loopControlDecoration = loopInst->findDecoration<IRLoopControlDecoration>())
                 {
-                    switch (loopControlDecoration->getMode())
+                    const auto loopMode = loopControlDecoration->getMode();
+                    if (loopMode == kIRLoopControl_Unroll)
                     {
-                    case kIRLoopControl_Unroll:
-                        // Note: loop unrolling control is only available in HLSL, not GLSL
-                        if(getSourceStyle() == SourceStyle::HLSL)
+                        switch (getSourceStyle())
                         {
-                            m_writer->emit("[unroll]\n");
+                            case SourceStyle::HLSL:
+                            {
+                                m_writer->emit("[unroll]\n");
+                                break;
+                            }
+                            case SourceStyle::GLSL:
+                            {
+                                // Note: loop unrolling control not available in GLSL
+                                break;
```

Comment:
It looks like there is actually a [GLSL extension](https://github.com/KhronosGroup/GLSL/blob/master/extensions/ext/GL_EXT_control_flow_attributes.txt) defined for control-flow-related attributes so that GLSL can generate the SPIR-V stuff. We should consider mapping the Slang attributes over to those and adding the extension as a requirement when they are used (assuming glslang supports them).

The gotcha here is that just passing down an unrolling hint to SPIR-V won't necessarily solve the problem that certain loops needs to be unrolled *before* GLSL/SPIR-V gets generated, or else the code won't conform to the constraints of the language spec (e.g., that certain function call arguments need to be compile-time constants).